components/dialog/dialog-mixin.js

```diff
@@ -177,10 +177,33 @@ export const DialogMixin = superclass => class extends RtlMixin(superclass) {
 		}
 	}
 
+	_findAutofocusElement(node) {
+		if (this._useNative) {
+			// Do not override native autofocus attribute implementation
+			return null;
+		}
+
+		if (!node.querySelector('slot')) {
+			// We are in a confirm dialog, autofocus attribute will never be set
+			return null;
+		}
+
+		const content = node.querySelector('slot').assignedElements({ flatten: true });
+		const el = content.find(el => el.autofocus)
+			?? content.find(el => el.querySelector('[autofocus]'))?.querySelector('[autofocus]');
```

Comment:
This part is messy. Because `assignedElements` returns an array instead of a node, I have to handle the case of the immediate (depth=0) element being autofocus-able separately from its children.

e.g.
```
[
  <p autofocus> ... </p>
]
```
won't have a querySelector method.

whereas:
```
[
  <div>
    <p autofocus> ... </p>
  </div>
]
```
will need to use querySelector.